lib/csv.rb

```diff
@@ -854,6 +854,13 @@ def initialize(message, line_number)
     end
   end
 
+  # The error thrown when the parser encounters invalid encoding in CSV.
+  class InvalidEncodingError < MalformedCSVError
+    def initialize(encoding, line_number)
```

Comment:
How about adding a reader for the target encoding?
```suggestion
    attr_reader :encoding
    def initialize(encoding, line_number)
      @encoding = encoding
```

Reply:
We can't get the target encoding by setting it to `@encoding`. (See the above suggestion.)

Reply:
oops!